carbon-design-system/carbon-for-ibm-dotcom
# fix(masthead): apply global styles updates based on Unified Cloud Masthead project

### Related Ticket(s)

DDS Ticket: N/A
HC JIRA Ticket: https://jsw.ibm.com/browse/HC-1679

### Description
- This PR implements new design enhancements to the dotcom and Cloud masthead that came out of the "Unified Cloud Masthead + Mega Menu Project" (see project overview doc - https://ibm.ent.box.com/notes/775640792107).
- We're using this PR to capture the overall design QA for the Cloud masthead + mega menu front-end contribution effort.
- Expect a fail in the Percy visual diffs as these changes are applied to the dotcom masthead container as well.

### Design Spec
- Masthead: https://ibm.ent.box.com/file/786859527229
- Mega Menu: https://ibm.ent.box.com/file/774092317795

### Changelog

Cloud Masthead
![Screen Shot 2021-04-28 at 10 19 06 AM](https://user-images.githubusercontent.com/1815714/116419595-35768b00-a80b-11eb-9e14-5863ede5d954.png)

Dotcom Masthead
![Screen Shot 2021-04-28 at 10 18 54 AM](https://user-images.githubusercontent.com/1815714/116419690-4e7f3c00-a80b-11eb-8671-f991ed93e68f.png)

- All of the design changes in this PR are based on the specs and docs that the design team has agreed upon (see doc - https://ibm.ent.box.com/notes/787865510080).

**Global Changes** 

- [x] All L0 menu item text reduced to 14px (from 16px) now using $body-short-01 
- [x] All drop down menu item text sizes reduced to 14px (from 16px) now using $body-short-01 
- [x] All L0 menu icons reduced to 16px (from 20px)
- [x] All L0 menu items using $text-02 for initial state (changes to $text-01 on hover and selected states)
- [x] All drop down menu items using $text-02 for initial state (changes to $text-01 on hover and selected states)
- [x] "Log in" text for logged out state of the menu (no drop down in logged out state)
- [x] User icon for account drop down once logged in (menu items will vary between IBM.com and Cloud)


**Testing Instructions**
- Visit the `Cloud Masthead` Storybook variant - `/?path=/story/components-cloud-masthead--default`
- Ensure that the masthead + mega menu components matche with the design specs for all breakpoints.
- Visit the `Dotcom Masthead` Storybook variant - `/?path=/story/components-masthead--default`
- Ensure that the masthead + mega menu components matche with the design specs for all breakpoints.
